### PR TITLE
Implement joint torque control device

### PR DESCRIPTION
**Pull request by isorrentino**
This PR integrates the `JointTorqueControlDevice` in BLF. The original implementation can be found in https://github.com/ami-iit/element_torque-control-via-current/tree/ae578859850ac83d8646a8536bb59999a6f60b8f/device/motorCurrentControl. I moved the device here to have it in a maintained repo.

The device introduces the possibility of running joint torque control on the top of the `ICurrentControl` interface or in general a control mode on top of another. Here is also an example of how to use the device on a real robot or on a fake robot.

Note that the PR has to be tested on the real robot.

### CODE REVIEW

**Comment by isorrentino** [comment]
What's happening to the Changelog Checker?

File: devices/JointTorqueControlDevice/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Copyright (C) 2020 Istituto Italiano di Tecnologia (IIT). All rights reserved.
+# This software may be modified and distributed under the terms of the
+# BSD-3-Clause license.
+
+if(FRAMEWORK_COMPILE_JointTorqueControlDevice)
+  # Warning: the <package> option of yarp_configure_plugins_installation should be different from the plugin name
```
Comment (traversaro) [inline devices/JointTorqueControlDevice/CMakeLists.txt:6]: Leftover?

**Review by traversaro — APPROVED** [review]

File: devices/JointTorqueControlDevice/CMakeLists.txt (hunk 1 above)
Comment (isorrentino) [reply devices/JointTorqueControlDevice/CMakeLists.txt:6]: Yes : )

**Comment by isorrentino** [comment]
Today I tested the device on the robot and it works fine. I'm going to add some modifications to this device but I'll merge them in a new PR.

**Comment by GiulioRomualdi** [comment]
So we can  merge this?

**Comment by isorrentino** [comment]
yes

**Comment by isorrentino** [comment]
Closing this PR, I will open a new PR containing the friction estimator through PINN in addition to this device.

**Pull request closed without merging**